pytorch_lightning/utilities/apply_func.py

```diff
@@ -257,7 +257,7 @@ def move_data_to_device(batch: Any, device: Union[str, torch.device]) -> Any:
 
     def batch_to(data: Any) -> Any:
         # try to move torchtext data first
-        if _TORCHTEXT_AVAILABLE and isinstance(data, Batch):
+        if _TORCHTEXT_AVAILABLE and not _TORCHTEXT_LEGACY_DEPRECATED and isinstance(data, Batch):
```

Comment:
I propose to emit a warning here, when user has a legacy Batch object, i.e., something like:

```py
rank_zero_warn("the torchtext.legacy.Batch object is deprecated and Lightning will remove support for it in v1.8. We recommend you to blablabla.")
```

Reply:
Why would we want that?

Lightning is supposed to abstract this from the user, and select the correct class depending on the version.

Reply:
My understanding is: There is no Batch class anymore in the latest torchtext. We want to remove the code entirely such that there is no torchtext Batch import anymore at all. In order to remove this support, we want to deprecate it such that users who user a newer version of Lightning and an older version of torchtext can make the transition.

@abhinavarora is my understanding about torchtext Batch correct?

Reply:
Hi @awaelchli, yes your understanding about torchtext Batch is correct.